pkg/parser/jsonschema.go

```diff
@@ -198,10 +197,6 @@ func (validator *JSONSchemaValidator) ValidateWithJSONSchema(rootNode *sitter.No
 //
 //	But in the JSON Schema, the `when` key is defined as an object, so the validation
```

Comment:
I'm wondering if this comment should be moved before the `if` instead. WDYT?

Reply:
Since the comment is long and the function is short, I think I'd rather have the purpose of the function explained, instead of having a block of text in the middle of the function. Additionally, the entire purpos of the function is to seal with this `when: << param >>` case.

A good rule of thumb I think would be:
* Outer comments document the why
* Inner comments document the how (and in should be avoided wherever possible in favor of readable, self-explanatory could)

https://www.jackfranklin.co.uk/blog/code-comments-why-not-how/

Up for debate, as this is more of a convention we can set for our teams/repos. Let's talk offline, merging as-is.